jgiven-core/src/main/java/com/tngtech/jgiven/impl/StageLifecycleManager.java

```diff
@@ -66,6 +72,33 @@ private void executeLifecycleMethods(LifecycyleMethodManager<?> register, boolea
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private void checkGuaranteedStatesAreInitialized() throws JGivenMissingGuaranteedScenarioStateException {
```

Comment:
have you thought about placing this functionality in the `ValueInjector` ? If so, why here?